packages/siwe/src/mapToSIWX.ts

```diff
@@ -19,37 +21,51 @@ export function mapToSIWX(siwe: AppKitSIWEClient): SIWXConfig {
     }
   }
 
-  ChainController.subscribeKey('activeCaipNetwork', async activeCaipNetwork => {
-    if (!siwe.options.signOutOnNetworkChange) {
-      return
-    }
+  async function signOut() {
+    await siwe.methods.signOut()
+    siwe.methods.onSignOut?.()
+  }
 
-    const session = await getSession()
-    const isDiffernetNetwork =
-      session &&
-      session.chainId !== NetworkUtil.caipNetworkIdToNumber(activeCaipNetwork?.caipNetworkId)
+  subscriptions.forEach(unsubscribe => unsubscribe())
+  subscriptions.push(
+    ChainController.subscribeKey('activeCaipNetwork', async activeCaipNetwork => {
+      if (!siwe.options.signOutOnNetworkChange) {
+        return
+      }
 
-    if (isDiffernetNetwork) {
-      await siwe.methods.signOut()
-    }
-  })
+      const session = await getSession()
+      const isDifferentNetwork =
+        session &&
+        session.chainId !== NetworkUtil.caipNetworkIdToNumber(activeCaipNetwork?.caipNetworkId)
 
-  ChainController.subscribeKey('activeCaipAddress', async activeCaipAddress => {
-    if (!siwe.options.signOutOnAccountChange) {
-      return
-    }
+      if (isDifferentNetwork) {
```

Comment:
I think you should only sign out the user if `signOutOnNetworkChange` is true

Reply:
There is a condition right above checking if `signOutOnNetworkChange` is false then it stops the function